tests/test_cfn_template.py

```diff
@@ -14,3 +14,17 @@ def test_init(self):
         template = templates["taskcat-json"]
         self.assertEqual(1, len(template.children))
         self.assertEqual(4, len(template.descendents))
+
+
+class TestCfnTemplateForEachResource(unittest.TestCase):
```

Comment:
This is essentially a copy-paste of the above but we just change out the dataset we are running against.

Sorry if this is redamentry. I am not a great test writer and defaulted back to just pattern matching. However, I do believe we have pretty complete coverage sans the implicit else on the [`if child_name`](https://github.com/aws-ia/taskcat/pull/845/files?show-viewed-files=true&file-filters%5B%5D=#diff-43923da38d7cc0dd443dc928f44bda163998ba332e653f7d81186042c182e2f9R169) statement.